main.tf

```diff
@@ -14,12 +14,14 @@ resource "google_monitoring_notification_channel" "clan_email" {
 }
 
 resource "google_monitoring_notification_channel" "clan_slack_channel" {
-  count        = var.clan_slack_channel != "" ? 1 : 0
+  for_each = var.clan_slack_channel
```

Comment:
The "count" and "for_each" meta-arguments are mutually-exclusive, only one should be used to be explicit about the number of resources to be created.